addons/l10n_in_edi/models/account_move.py

```diff
@@ -225,7 +230,35 @@ def _l10n_in_edi_send_invoice(self):
                         "doc_date": self.invoice_date and self.invoice_date.strftime("%d/%m/%Y"),
                     }
                 )
-                if not response.get("error"):
+                mismatch_error = []
+                if jwt:
+                    try:
+                        if SignedInvoice := response.get('data', {}).get('SignedInvoice', {}):
+                            decoded_response = jwt.decode(SignedInvoice, options={'verify_signature': False})
+                            decoded_response = json.loads(decoded_response.get('data', '{}'))
+                            received_gstin = decoded_response.get('BuyerDtls', {}).get('Gstin')
+                            expected_gstin = generate_json.get('BuyerDtls', {}).get('GSTIN')
+                            received_total_invoice_value = decoded_response.get('ValDtls', {}).get('TotInvVal')
+                            expected_total_invoice_value = generate_json.get('ValDtls', {}).get('TotInvVal')
+                            # Check for mismatch between decoded and expected e-invoice details:
+                            # - Buyer GSTIN must match
+                            # - Total Invoice Value must be within the allowed government tolerance range:
+                            #   For example, if the expected invoice value is 100,
+                            #   the valid range is from 99 (value - 1) to 101 (value + 1),
+                            #   i.e., 99.00 < received value < 101.00
+                            if (received_gstin != expected_gstin or
+                                not expected_total_invoice_value - 1 < received_total_invoice_value < expected_total_invoice_value + 1
+                            ):
+                                mismatch_error = [{
+                                    'code': '2150',
+                                    'message': _("Duplicate IRN found for this invoice, but the buyer details or invoice values do not match.")
+                                }]
+                    except (json.JSONDecodeError, jwt.exceptions.DecodeError) as e:
```

Comment:
this try except should only be around the json.loads because that is what it checks...